src/feature/hs/hs_config.c

```diff
@@ -471,25 +475,23 @@ config_generic_service(const config_line_t *line_,
   int have_dir_group_read = 0, have_max_streams = 0;
   int have_max_streams_close = 0;
 
-  tor_assert(line_);
+  tor_assert(line);
   tor_assert(options);
   tor_assert(service);
 
   /* Makes thing easier. */
   config = &service->config;
 
+  if (BUG(strcasecmp(line->key, "HiddenServiceDir")))
+    return -1;
+
   /* The first line starts with HiddenServiceDir so we consider what's next is
    * the configuration of the service. */
-  for (line = line_; line ; line = line->next) {
+  for ( ; line ; line = line->next) {
     int ok = 0;
 
     /* This indicate that we have a new service to configure. */
     if (!strcasecmp(line->key, "HiddenServiceDir")) {
```

Comment:
Not sure this if is needed anymore. Reason is that we have the `if (BUG())` before so we know that when we enter that function, the start is the `HiddenServiceDir` so we could move the `config->directory_path = tor_strdup(line->value);` below even at the top after the check.

Reply:
I think this line is also gone in the final version?